create-ava-rule.js

```diff
@@ -82,11 +82,9 @@ module.exports = () => {
 			}
 		},
 		CallExpression: node => {
-			if (!currentTestNode) {
```

Comment:
It was a safe-guard (or rather optimization?) to prevent entering a new test when you were already in one.

It will only be able to make a difference for other rules (but probably won't) if you have nested tests. I can improve it to make sure the other rules won't be affected, but I'm not sure that it's worth it as there is a hard stance against nested tests. Let me know what you prefer.